h2o-algos/src/main/java/hex/word2vec/WordVectorTrainer.java

```diff
@@ -1,124 +1,80 @@
 package hex.word2vec;
 
-import water.H2O;
+import water.DKV;
+import water.Key;
 import water.MRTask;
-import water.fvec.CStrChunk;
-import water.fvec.Vec;
 import water.fvec.Chunk;
-import water.fvec.Frame;
-import water.nbhm.NonBlockingHashMap;
 import water.parser.BufferedString;
-import water.util.Log;
 import hex.word2vec.Word2VecModel.*;
-import hex.word2vec.Word2Vec.*;
-import java.util.Random;
+import water.util.ArrayUtils;
+import water.util.IcedHashMap;
+import water.util.IcedHashMapGeneric;
+import water.util.IcedLong;
+
+import java.util.Iterator;
 
 public class WordVectorTrainer extends MRTask<WordVectorTrainer> {
```

Comment:
I will make a comment about the structure of _syn0 and _syn1, is that what you meant?